lib/krane/deploy_task.rb

```diff
@@ -285,17 +284,21 @@ def validate_configuration(prune:)
     measure_method(:validate_configuration)
 
     def validate_resources(resources)
-      validate_globals(resources)
-      batch_dry_run_success = validate_dry_run(resources)
       resources.select! { |r| r.selected?(@selector) } if @selector_as_filter
-      Krane::Concurrency.split_across_threads(resources) do |r|
+      validate_globals(resources)
+      applyables, individuals = resources.partition { |r| r.deploy_method == :apply }
+      batch_dry_run_success = validate_dry_run(applyables)
+      resources_to_validate = batch_dry_run_success ? individuals : resources
+
+      Krane::Concurrency.split_across_threads(resources_to_validate) do |r|
```

Comment:
Originally I wanted to to just trust the output of the batched dry run test and raise a `FatalDeploymentError`. However, that made it **very** difficult to give a consistent UX with regards to giving useful feedback: the main issue being it became necessary to suppress all feedback if a sensitive document/secret was contained in the apply set.

To that end, I've kept things pretty similar, just removing resources from the per-resource checks if batched dry-run succeeds. If it fails, since that's a less common case, we default to per-resource checking in order to provide more granular feedback and highlight what the actual issue it. This is the difference between "Something somewhere went wrong" and "THIS particular file and THIS particular line are wrong".